core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -162,10 +160,10 @@ protected Expression rowFilter() {
 
   protected List<DataFile> addedDataFiles() {
     return ImmutableList.copyOf(
```

Comment:
we can also just collect directly using `ImmutableList.toImmutableList()` to achieve the same thing so that we don't do the extra copy. I've updated that